interp/notation.ml

```diff
@@ -760,16 +760,20 @@ let rebuild_arguments_scope (req,r,n,l,_) =
   match req with
     | ArgsScopeNoDischarge -> assert false
     | ArgsScopeAuto ->
-        let scs,cls = compute_arguments_scope_full (fst(Global.type_of_global_in_context (Global.env ()) r)(*FIXME?*)) in
-	(req,r,List.length scs,scs,cls)
+      let sigma, env = Evd.empty, Global.env () in (*FIXME?*)
+      let typ = EConstr.of_constr @@ fst (Global.type_of_global_in_context env r) in
+      let scs,cls = compute_arguments_scope_full sigma typ in
+      (req,r,List.length scs,scs,cls)
     | ArgsScopeManual ->
-	(* Add to the manually given scopes the one found automatically
-           for the extra parameters of the section. Discard the classes
-           of the manually given scopes to avoid further re-computations. *)
-        let l',cls = compute_arguments_scope_full (fst (Global.type_of_global_in_context (Global.env ()) r)) in
-	let l1 = List.firstn n l' in
-        let cls1 = List.firstn n cls in
-	(req,r,0,l1@l,cls1)
+      (* Add to the manually given scopes the one found automatically
+         for the extra parameters of the section. Discard the classes
+         of the manually given scopes to avoid further re-computations. *)
+      let sigma, env = Evd.empty, Global.env () in (*FIXME?*)
+      let typ = EConstr.of_constr @@ fst (Global.type_of_global_in_context env r) in
+      let l',cls = compute_arguments_scope_full sigma typ in
```

Comment:
The branches do the same thing up to this point, ie the function is equivalent to
```ocaml
let rebuild_arguments_scope (req,r,n,l,_) =
  let sigma, env = Evd.empty, Global.env () in
  (*FIXME?*)
  let typ = EConstr.of_constr @@ fst (Global.type_of_global_in_context env r) in
  let scs,cls = compute_arguments_scope_full sigma typ in
  match req with
  | ArgsScopeNoDischarge -> assert false
  | ArgsScopeAuto ->
    (req,r,List.length scs,scs,cls)
  | ArgsScopeManual ->
    (* Add to the manually given scopes the one found automatically
       for the extra parameters of the section. Discard the classes
       of the manually given scopes to avoid further re-computations. *)
    let l1 = List.firstn n scs in
    let cls1 = List.firstn n cls in
    (req,r,0,l1@l,cls1)
```
Shall we do this combination?

Reply:
If @herbelin is OK I'm all for it.